calyx-opt/src/analysis/compute_static.rs

```diff
@@ -82,13 +82,9 @@ impl WithStatic for ir::Seq {
     type Info = CompTime;
     fn compute_static(&mut self, extra: &Self::Info) -> Option<u64> {
         // Go through each stmt in the seq, and try to calculate the latency.
-        self.stmts.iter_mut().fold(Some(0), |acc, stmt| {
-            match (acc, stmt.update_static(extra)) {
-                (Some(cur_latency), Some(stmt_latency)) => {
-                    Some(cur_latency + stmt_latency)
-                }
-                (_, _) => None,
-            }
+        self.stmts.iter_mut().try_fold(0, |cur_latency, stmt| {
+            stmt.update_static(extra)
+                .map(|stmt_latency| cur_latency + stmt_latency)
```

Comment:
This might potentially be the cause of the compiler failures. I'm not sure if the `try_fold` is computing the same thing as before

Reply:
Specifically, I was seeing the same error when I made these changes: https://github.com/calyxir/calyx/pull/1921/commits/a4b65337a0818f352a299abee31bc29290fdca0c

Reply:
Ah, I see the problem. This is tricky code. Even when the accumulator is `None`, the code will go through the rest of the statements in the `seq` and run `update_static` on them. In contrast, `try_fold` implements an early exit when the accumulator is None.